sdk/servicebus/service-bus/src/dataTransformer.ts

```diff
@@ -47,40 +52,159 @@ export const defaultDataTransformer = {
-        processedBody = JSON.parse(bodyStr);
-      } catch (err) {
-        logger.verbose(
-          "[decode] An error occurred while trying JSON.parse() on the received body. " +
-            "The error is %O",
-          err
-        );
+      if (isRheaAmqpSection(body)) {
+        switch (body.typecode) {
+          case dataSectionTypeCode:
+            return { body: tryToJsonDecode(body.content), bodyType: "data" };
+          case sequenceSectionTypeCode:
+            // typecode:
+            // handle sequences
+            return { body: body.content, bodyType: "sequence" };
+          case valueSectionTypeCode:
+            // value
+            return { body: body.content, bodyType: "value" };
+        }
+      } else {
+        // not sure - we have to try to infer the proper bodyType and content
+        if (isBuffer(body)) {
+          // This indicates that we are getting the AMQP described type. Let us try decoding it.
+          return { body: tryToJsonDecode(body), bodyType: "data" };
+        } else {
+          return { body: body, bodyType: "value" };
+        }
       }
     } catch (err) {
       logger.verbose(
         "[decode] An error occurred while decoding the received message body. The error is: %O",
         err
       );
+      throw err;
     }
-    return processedBody;
   }
+
+  /**
+   * A function that takes the body property from an AMQP message
+   * (an AMQP Data type (data section in rhea terms)) and returns the decoded message body.
+   * If it cannot decode the body then it returns the body
+   * as-is.
+   * @param body - The AMQP message body
+   * @returns decoded body or the given body as-is.
+   */
+  // decode(body: unknown): any {
+  //   let processedBody: any = body;
+  //   try {
+  //     if (isObjectWithProperties(body, ["content"]) && isBuffer(body.content)) {
+  //       // This indicates that we are getting the AMQP described type. Let us try decoding it.
+  //       processedBody = body.content;
+  //     }
+  //     try {
+  //       // Trying to stringify and JSON.parse() anything else will fail flat and we shall return
+  //       // the original type back
+  //       const bodyStr: string = processedBody.toString("utf8");
+  //       processedBody = JSON.parse(bodyStr);
+  //     } catch (err) {
+  //       logger.verbose(
+  //         "[decode] An error occurred while trying JSON.parse() on the received body. " +
+  //           "The error is %O",
+  //         err
+  //       );
+  //     }
+  //   } catch (err) {
+  //     logger.verbose(
+  //       "[decode] An error occurred while decoding the received message body. The error is: %O",
+  //       err
+  //     );
+  //   }
+  //   return processedBody;
+  // }
 };
+
+/** @internal */
+export function isRheaAmqpSection(
+  possibleSection: any | RheaAmqpSection
+): possibleSection is RheaAmqpSection {
+  return possibleSection != null && typeof possibleSection.typecode === "number";
```

Comment:
Shouldnt it also have a "content"?

Reply:
Do you mean this?
```js
/** @internal */
export function isRheaAmqpSection(
  possibleSection: any | RheaAmqpSection
): possibleSection is RheaAmqpSection {
  return (
    possibleSection != null &&
    typeof possibleSection.typecode === "number" &&
    isDefined(possibleSection.content)
  );
}
```

Reply:
Updated to that.. let me know what you think.